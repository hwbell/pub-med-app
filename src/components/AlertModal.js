import React from 'react';
import '../App.scss';
import 'bootstrap/dist/css/bootstrap.css';

// components
import { Button, Input, Form, InputGroup, Modal, ModalHeader, ModalBody, ModalFooter } from 'reactstrap';

// this form is toggled visible by the 'add to collection' button on the search page
class AlertModal extends React.Component {
  constructor(props) {
    super(props);
  }

  componentDidMount() {
    // setTimeout()
  }

  render() {

    return (
      <Modal centered={true}
        isOpen={this.props.isVisible}
        toggle={this.props.toggle}>

        {/* <ModalHeader toggle={this.props.toggle}>Your Collections</ModalHeader> */}
        <ModalBody>

          <p style={styles.articleText}>{this.props.message}</p>

        </ModalBody>
        <ModalFooter>
          <Button color="primary" size="sm" onClick={this.props.confirm}>OK</Button>
          {this.props.confirming &&
            <Button color="secondary" size="sm" onClick={this.props.toggle}>cancel</Button>
          }
        </ModalFooter>

      </Modal>

    );
  }
}

const styles = {
  content: {
    height: '100%',
  },
  articleText: {
    color: 'black',
    margin: '10px 20px',
    fontSize: '18px'
  },

}

export default AlertModal;
