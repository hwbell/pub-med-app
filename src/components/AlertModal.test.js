import React from 'react';
import AlertModal from './AlertModal';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';

const toggleStub = jest.fn();

describe('AlertModal', () => {

  let someProps;
  let wrapper, instance;
  beforeAll(async () => {
    someProps = {
      message: 'Hey! You have to provide a unique name!',
      toggle: toggleStub,
      isVisible: false
    }
  });

  beforeEach( () => {
    wrapper = shallow(<AlertModal {...someProps} />)
    instance = wrapper.instance();
  })

  // tests
  it('renders without crashing', async () => {
    let wrapper = shallow(<AlertModal {...someProps}/>);
    // wrapper.update();

  });

  it('renders correctly', async () => {
    const tree = await renderer
      .create(<AlertModal {...someProps}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders the correct elements', () => {
    ['Modal', 'ModalBody', 'ModalFooter', {color: "primary"}].forEach((selector) => {
      expect(wrapper.find(selector).length).toEqual(1);
    })
  })

})

