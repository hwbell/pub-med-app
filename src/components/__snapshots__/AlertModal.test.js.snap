// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertModal renders correctly 1`] = `null`;
